erpnext/accounts/report/gross_profit/gross_profit.py

```diff
@@ -477,6 +479,17 @@ def process(self):
 		if self.grouped:
 			self.get_average_rate_based_on_group_by()
 
+	def has_delivery_notes(self, row):
+		for x in self.delivery_notes:
```

Comment:
It will go through all the delivery notes and items for each and every Sales Invoice/Row. Could be a heavy operation for users who have a lot of Delivery Notes. Can we make some sort of map to check this?

Reply:
Created a Map.